src/components/DisplayMenuAndOrder.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { Component } from "react";
+import { getData } from "../modules/productData";
+import axios from "axios";
+import { authenticate } from "../modules/authenticate";
+
+class DisplayMenuAndOrder extends Component {
+  state = {
+    productData: [],
+    message: {},
+    orderDetails: {},
+    showOrder: false
+  };
+
+  componentDidMount() {
+    this.getProductData();
+  }
+
+  async getProductData() {
+    let result = await getData();
+    this.setState({ productData: result.data.products });
+  }
+
+  async addToOrder(event) {
+    let id = event.target.parentElement.dataset.id;
+    let result;
+    if (this.state.orderDetails.hasOwnProperty("id")) {
+      result = await axios.put(
+        `http://localhost:3000/api/orders/${this.state.orderDetails.id}`,
+        { product_id: id }
```

Comment:
I see that you have a user logged in (in your tests) How do you communicate that to the backend server and I what way is it relevant

Reply:
We don't fully understand your question about in what way it communicates to the backend server, but in regards to the purpose of the tests, we added the functionality that the user can only see the menu and create order if they are logged in. So we had to update the tests to login user before testing the order and display menu functionality.

Reply:
Okay, thx